locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/softlayer/SoftLayerSameVlanLocationCustomizer.java

```diff
@@ -0,0 +1,398 @@
+import org.apache.brooklyn.location.jclouds.BasicJcloudsLocationCustomizer;
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
+import org.apache.brooklyn.location.jclouds.JcloudsLocationCustomizer;
+import org.apache.brooklyn.location.jclouds.JcloudsMachineLocation;
+import org.apache.brooklyn.location.jclouds.JcloudsSshMachineLocation;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.flags.SetFromFlag;
+import org.apache.brooklyn.util.core.task.Tasks;
+import org.apache.brooklyn.util.text.Strings;
+import org.apache.brooklyn.util.time.Duration;
+
+/**
+ * Customizes {@link JcloudsSshMachineLocation machines} in SoftLayer to use
+ * the same VLANs across an application, or other named scope
+ * <p>
+ * Define the scope by setting the {@link #SCOPE_UID scopeUid} ({@code softlayer.vlan.scopeUid})
+ * option. If unset or blank, will default to the {@link Entity#getApplicationId() application ID}
+ * of the entity being deployed on the machine. Set {@link #SCOPE_TIMEOUT scopeTimeout}
+ * ({@code softlayer.vlan.timeout}) to change the length of time the customizer
+ * will wait for VLAN information; normally ten minutes.
+ * <p>
+ * The VLAN IDs and latches are stored as {@link ConfigKey configuration} on
+ * the {@link JcloudsLocation location} provisioning the VMs, in a map keyed
+ * on the scope.
+ */
+@ThreadSafe
+public class SoftLayerSameVlanLocationCustomizer extends BasicJcloudsLocationCustomizer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SoftLayerSameVlanLocationCustomizer.class);
+
+    @SetFromFlag("scopeUid")
+    public static final ConfigKey<String> SCOPE_UID = ConfigKeys.newStringConfigKey(
+            "softlayer.vlan.scopeUid",
+            "The unique identifier for a Softlayer location scope that will have VMs created in the same VLAN");
+
+    @SetFromFlag("scopeTimeout")
+    public static final ConfigKey<Duration> SCOPE_TIMEOUT = ConfigKeys.newDurationConfigKey(
+            "softlayer.vlan.timeout",
+            "The length of time to wait for a Softlayer VLAN ID",
+            Duration.minutes(15));
+
+    public static final ConfigKey<Map<String, CountDownLatch>> COUNTDOWN_LATCH_MAP = ConfigKeys.newConfigKey(
+            new TypeToken<Map<String, CountDownLatch>>() { },
+            "softLayerSameVlanLocationCustomizer.map.latches",
+            "A mapping from scope identifiers to CountDownLatches; used to synchronize threads",
+            MutableMap.<String, CountDownLatch>of());
+    public static final ConfigKey<Map<String, Integer>> PUBLIC_VLAN_ID_MAP = ConfigKeys.newConfigKey(
+            new TypeToken<Map<String, Integer>>() { },
+            "softLayerSameVlanLocationCustomizer.map.publicVlanIds",
+            "A mapping from scope identifiers to public VLAN numbers",
+            MutableMap.<String, Integer>of());
+    public static final ConfigKey<Map<String, Integer>> PRIVATE_VLAN_ID_MAP = ConfigKeys.newConfigKey(
```

Comment:
i think it would simplify things a lot to define an object with

    transient CountdownLatch latch;
    Integer publicVlanId;
    Integer privateVlanId;

and then a single map capturing this, and then lookup/save this object rather than the ID's and latches separately.  this also allows us to:

* make the countdown transient which i think is needed for when locations are persisted
* combine the various lookup/save methods for vlan id's